tools/variables/_all.ps1

```diff
@@ -12,9 +12,35 @@ param (
 
 $vars = @{}
 
-Get-ChildItem "$PSScriptRoot\*.ps1" -Exclude "_*" |% {
+Get-ChildItem "$PSScriptRoot\*.ps1" -Exclude "_*" | % {
     Write-Host "Computing $($_.BaseName) variable"
-    $vars[$_.BaseName] = & $_
+    # AzDO variables are stored as environment variables which has a max length of ~32k characters.
+    if ($_.Length -gt 32000) {
+        # If the InsertJsonValues script returns a value longer than that,
+        # write the value to a temp file and use the InsertJsonValuesFile variable that InsertVSPayload understands.
+        if ($_.BaseName -eq "InsertJsonValues") {
+            Write-Warning "The script InsertJsonValues.ps1 is longer than 32k characters."
+            Write-Warning "Writing to a temp file and setting INSERTJSONVALUESFILE to the location of the file."
```

Comment:
Rather than special case InsertJsonValues in _this_ script (which should be general purpose), I think converting the value to a .txt file should be done at insertion time where it has to deal with this variable specifically.

Reply:
So you're saying that the check (and converting it to a text file) should be in _define.ps1 ?